jupyterhub_traefik_proxy/etcd.py

```diff
@@ -63,8 +64,28 @@ def _default_client(self):
         help="""the etcd key prefix for traefik configuration""",
     )
 
+    traefik_api_url = Unicode(
+        "http://127.0.0.1:8099",
+        config=True,
+        help="""traefik authenticated api endpoint url""",
+    )
+
+    traefik_api_basic_auth_hashed_password = Unicode(
+        "$apr1$LD7Rp6hQ$QzRELO5C153vGLFIDorKJ0",
+        config=True,
+        help="""traefik api basic authentication password""",
+    )
+
+    traefik_api_basic_auth_password = Unicode(
+        "admin", config=True, help="""traefik api basic authentication password"""
```

Comment:
Let's not have this defined by default. Instead, in the tests create the Proxy with:

```python
proxy = TraefikEtcdProxy(traefik_api_basic_auth_password='...')
```

in the test. We could also probably call this `traefik_api_password` without `basic_auth`, since that's implied. Same for username.